src/txmempool.cpp

```diff
@@ -686,13 +685,11 @@ void CTxMemPool::remove(const CTransactionBase& origTx, std::list<CTransaction>&
 
             const uint256& scid = cert.GetScId();
 
-#ifdef ENABLE_ADDRESS_INDEXING
             // are we removing a top-quality cert?
             const uint256& topQualHash = mapSidechains.at(scid).GetTopQualityCert()->second;
             CSidechain sidechain;
             assert(pcoinsTip->GetSidechain(scid, sidechain));
             bool isTopQualityCert = (topQualHash == hash) || sidechain.isNonCeasing();
-#endif // ENABLE_ADDRESS_INDEXING
```

Comment:
Ok, seems reasonable, modified accordingly.

Reply:
Looks fine to me!

Reply:
Reverted this because of failure on `sc_cert_addressindex.py` and `sc_cert_addrmempool.py`.
Analysing the code and inspecting some logs, I'd say the problem with proposed change is that the `GetTopQualityCert` call is moved after `EraseCert` call; I see occurrences where the original top quality cert is the one being erased, hence the new top quality cert is different from the original, leading to some unexpected behavior.